src/schema/v2/previewSavedSearch.ts

```diff
@@ -99,13 +101,70 @@ const PreviewSavedSearchType = new GraphQLObjectType<any, ResolverContext>({
       type: new GraphQLNonNull(
         new GraphQLList(new GraphQLNonNull(SearchCriteriaLabel))
       ),
-      resolve: (_) => {
-        return mockSuggestedFilters
+      resolve: async (_root, _args, { filterArtworksLoader }) => {
+        const suggestedFiltersByArtist: SearchCriteriaLabel[][] = await Promise.all(
+          _root.artistIDs.map((artistSlug) =>
+            getSuggestedFiltersByArtistSlug(artistSlug, filterArtworksLoader)
+          )
+        )
+
+        return _.chain(suggestedFiltersByArtist)
+          .flatten()
+          .uniqBy((searchCriteria) => JSON.stringify(searchCriteria))
+          .value()
       },
     },
   }),
 })
 
+const getMostPopularField = (aggregation: {
+  [key: string]: { name: string; count: number }
+}): string => {
+  return Object.values(aggregation).reduce((acc, curr) => {
+    return curr.count > acc.count ? curr : acc
+  }).name
+}
+
+const getRaritySearchCriteriaLabel = (
+  name: string | undefined
+): SearchCriteriaLabel | null => {
+  if (!name || name === "unknown edition") {
+    return null
+  }
+
+  return {
+    displayValue: attributionClasses[name].name,
+    field: "attributionClass",
+    value: name,
+    name: "Rarity",
+  }
+}
+
+const getSuggestedFiltersByArtistSlug = async (
+  artistSlug: string,
+  filterArtworksLoader
+): Promise<SearchCriteriaLabel[]> => {
+  const gravityArgs = {
+    published: true,
+    aggregations: ["attribution_class"],
+    artist_id: artistSlug,
+  }
+
+  const { aggregations } = await filterArtworksLoader(gravityArgs)
```

Comment:
**suggestion:** In cases where are interested _only_ in the aggregations but _not_ in the artwork results, we can also, specify `size: 0` to reduce the speed/size of the response from gravity. 

Example: https://github.com/artsy/metaphysics/blob/7cdcd8ba40a28bdf39c4dcc310be6f19439634de/src/schema/v2/artists/curatedTrending.ts#L11-L17